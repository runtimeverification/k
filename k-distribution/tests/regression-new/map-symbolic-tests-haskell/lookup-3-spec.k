// Copyright (c) Runtime Verification, Inc. All Rights Reserved.
// generated by gen-tests.sh

requires "map-tests.k"

module LOOKUP-3-SPEC
    imports MAP-TESTS

    claim <k> lookup ( (MAP:Map X:MyId |-> 1) [Y:MyId <- 2] [ Z:MyId ] ) => . </k> requires Y ==K X andBool Z =/=K Y andBool Z =/=K X

endmodule
