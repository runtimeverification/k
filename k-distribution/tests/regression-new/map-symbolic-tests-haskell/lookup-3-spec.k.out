  #Not ( {
    Y
  #Equals
    Z
  } )
#And
  <k>
    lookupResult ( MAP [ Z:MyId ] ~> . ) ~> .
  </k>
#And
  {
    false
  #Equals
    Y:MyId in_keys ( MAP )
  }
#And
  {
    true
  #Equals
    Z:MyId in_keys ( MAP
    ( Y:MyId |-> 2 ) )
  }
#And
  {
    true
  #Equals
    Z:MyId in_keys ( MAP )
  }
